MessageManager.class.nut

```diff
@@ -792,8 +797,10 @@ class MessageManager {
             _isFunc(msg._onAck) && msg._onAck(msg);
             _isFunc(_onAck) && _onAck(msg);
 
-            // Delete the acked message from the queue
-            delete _sentQueue[id];
+            // Delete the acked message from the queue if there is no _onReply handler set (either global or message-specific)
```

Comment:
Yes, it will. But if a message was ACKed why should it timeout? I'm looking into it right now. Let me fix it on your branch (I hope I have rights to do so).